### ISSUE TITLE
Uncaught TypeError: Cannot read property 'params' of null @ start of touch event

### ISSUE THREAD

**Issue by maillig** [zefoy/ngx-swiper-wrapper#17]
The following error occurs when you use the wrapper on mobile devices or the viewports of browsers like Chrome: 

Uncaught TypeError: Cannot read property 'params' of null
    at HTMLDivElement.x.onTouchStart (app.js:640)
    at t.invokeTask (app.js:975)
    at e.runTask (app.js:975)
    at HTMLDivElement.invoke (app.js:975)

The above error appears in the provided example: [https://zefoy.github.io/ngx-swiper-wrapper/].

**Comment by sconix** [comment]
I have only seen that in browser touch emulation state, although I haven't checked on actual mobile device. Its a bug in the Swiper library itself and I made a fix at some point for it, but then I noticed that the error was not critical and did not make a PR. I hope that the next Swiper release would fix that error if not I can make the fix again and make a PR to the Swiper project.

**Comment by maillig** [comment]
I see, 

Thank you for the clarification. Good to know its not a critical error. If possible , I would appreciate the fix in a PR till Swiper solves the bug.

**Comment by sconix** [comment]
I will definitely look into this next month since I need to fix this somehow before we launch our new product since this error exists in our app as well.

**Comment by Dranix** [comment]
I have the similar error when trying to use this library with Ionic 2. Any work around until sconix patch it?

**Comment by snortblt** [comment]
Seeing the same issue with [config] here. This is a deal-breaker for me because I need a vertical swiper nested in horizontal, so I can't fall back to a single global config. Any status update on this?

**Comment by sconix** [comment]
If you can tell me how to reproduce the problem with config I can fix it. The params is a bug in Swiper and there is an issue about that and it not related to this library nor the config issue you are seeing.

**Comment by JanMesaric** [comment]
I have encountered the same problem. Will test on actual device, but this fails in chrome using v3.4.2.

**Comment by JanMesaric** [comment]
I'm getting the same error when using Swiper on its own when destroying and reiniting. I solved it by overriding the first parameter in destroy() method, setting it to false. 
`
deleteInstance - boolean - Set it to false (by default it is true) to not to delete Swiper instance`

After this I manually remove DOM and successfully create a memory leak 👎 

Not sure how to solve this properly.

**Comment by sconix** [comment]
Yes there is some bug in the Swiper destroy method. Lets hope that it would be fixed in next release.

**Comment by dtslvr** [comment]
@Dranix I have the same issue with Ionic 4. As soon as I add a config (with e.g. `direction: 'horizontal'`) I get `Uncaught TypeError: Cannot read property 'params' of null at HTMLDivElement.Swiper.s.onTouchStart (swiper.js:1544)`. Any news or should I also try to patch `swiper.js`?

**Comment by sconix** [comment]
There are no new news from the Swiper project. There is a really easy way to patch it, you just need to add if (this.swiper) check before the line where the error happens. But thats just a quick fix, the proper fix would be to tear down the event listener properly on destroy. I try to make a proper fix and PR to Swiper during this week or at the latest on next week.

**Comment by sconix** [comment]
I looked into this and could not find a bug in Swiper. There was very odd situation that the event listener was removed correctly but it did not remove. My conclusion is that something has to conflict with Swiper/browsers event handling. After that I noticed that with fresh npm install the error was removed from the example app at least. So what ever it was that caused the error has been fixed in some package afaik. So I am closing this for now and if somebody can still reproduce this with latest versions of the dependencies then I can open this again.
